cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/InMemoryProgramRunnerModule.java

```diff
@@ -93,32 +95,29 @@ protected void configure() {
 
     // Bind ProgramStateWriter
     bind(ProgramStateWriter.class).to(DirectStoreProgramStateWriter.class);
-    // For programs with multiple instances, disable the program state writer on the instances so that the higher
-    // controller can record the program state based on all of the instances
-    // TODO when CDAP-12179 is resolved, the ProgramStateWriter will be in the InMemoryProgramRunner, so there
-    // will be no special case
-    bind(ProgramStateWriter.class)
-      .annotatedWith(Names.named("programStateWriter"))
-      .to(NoOpProgramStateWriter.class);
 
     // Bind ProgramRunner
     MapBinder<ProgramType, ProgramRunner> runnerFactoryBinder =
       MapBinder.newMapBinder(binder(), ProgramType.class, ProgramRunner.class);
-    runnerFactoryBinder.addBinding(ProgramType.FLOW).to(FlowProgramRunner.class);
-    runnerFactoryBinder.addBinding(ProgramType.MAPREDUCE).to(MapReduceProgramRunner.class);
-    runnerFactoryBinder.addBinding(ProgramType.WORKFLOW).to(WorkflowProgramRunner.class);
-    runnerFactoryBinder.addBinding(ProgramType.WEBAPP).to(WebappProgramRunner.class);
+    runnerFactoryBinder.addBinding(ProgramType.FLOW).to(InMemoryFlowProgramRunner.class);
```

Comment:
This is needed so that in `DefaultProgramRunnerFactory`, we get the in-memory runners.